tools/psa/build_tfm.py

```diff
@@ -160,22 +181,301 @@ def clean_up_cloned_repos():
+    """
+    return [str(t) for t in TARGET_NAMES if
+            Target.get_target(t).is_TFM_target]
+
+def get_target_info(target, toolchain=None):
+    """
+    Creates a TF-M target tuple with default toolchain and
+    artifact delivery directory.
+
+    :param target: Target name.
+    :param toolchain: Toolchain
+    :return: tuple (target, toolchain, delivery directory).
+    """
+    if toolchain:
+        if not TARGET_MAP[target].tfm_supported_toolchains:
+            msg = "Supported Toolchains is not configured for target %s" % (
+                                                    TARGET_MAP[target].name)
+            raise Exception(msg)
+        elif toolchain not in TARGET_MAP[target].tfm_supported_toolchains:
+            msg = "Toolchain %s is not supported by %s" % (toolchain,
+                                                    TARGET_MAP[target].name)
+            raise Exception(msg)
+        tc = toolchain
+    else:
+        tc = TARGET_MAP[target].tfm_default_toolchain
+
+    delivery_dir = join(ROOT, 'targets',
+                        TARGET_MAP[target].delivery_dir)
+
+    if not os.path.exists(delivery_dir):
+        msg = "Delivery directory (delivery_dir) missing for %s" % target
+        raise FileNotFoundError(msg)
+
+    bl_sup = True if TARGET_MAP[target].tfm_bootloader_supported else False
+    return tuple([TARGET_MAP[target].name,
+                  TARGET_MAP[target].tfm_target_name,
+                  tc,
+                  delivery_dir,
+                  bl_sup])
+
+def get_mbed_supported_tfm_targets():
+    tfm_secure_targets = get_tfm_secure_targets()
+    logger.debug("Found the following TF-M targets: {}".format(
+                                                ', '.join(tfm_secure_targets)))
+
+    return [get_target_info(t) for t in tfm_secure_targets]
+
+def commit_changes(directory, target=None, toolchain=None):
+    """
+    Check for changes in `directory` and if any then commit them
+    :param directory: path to be checked for changes
+    :param target: Target name
+    :param toolchain: Toolchain name
+    """
+    # Use --intent-to-add option of git status to identify un-tracked files
+    cmd = ['git', '-C', ROOT, 'status', 'N', directory]
+    run_cmd_and_return_errorcode(cmd)
+
+    cmd = ['git', '-C', ROOT, 'diff', '--exit-code', '--quiet', directory]
+    changes_made = run_cmd_and_return_errorcode(cmd)
+
+    if changes_made:
+        if target:
+            logger.info("Change in image for %s has been detected" % target)
+        else:
+            logger.info("Change detected in directory %s" % directory)
+        cmd = ['git', '-C', ROOT, 'add', relpath(directory, ROOT)]
+        run_cmd_and_return_errorcode(cmd)
+
+        if target:
+            logger.info("Committing image for %s" % target)
+            msg = '--message="Updated secure binary for %s (%s)"' % (target,
+                                                                    toolchain)
```

Comment:
We need a space before `toolchain` to align it with `target`.

Reply:
Fixed